src/common/com/intellij/plugins/haxe/model/HaxeMethodModel.java

```diff
@@ -131,11 +128,19 @@ public String getPresentableText(HaxeMethodContext context) {
   }
 
   public SpecificFunctionReference getFunctionType() {
-    LinkedList<SpecificTypeReference> args = new LinkedList<SpecificTypeReference>();
+    return getFunctionType(null);
+  }
+
+  public ResultHolder getReturnType(@Nullable HaxeGenericResolver resolver) {
+    return HaxeTypeResolver.getFieldOrMethodReturnType((AbstractHaxeNamedComponent)this.getPsi(), resolver);
+  }
+
+  public SpecificFunctionReference getFunctionType(@Nullable HaxeGenericResolver resolver) {
```

Comment:
Small nit.  Please keep functions with identical names next to each other.  It just makes them easier to scan for.  In this case, getReturnType(resolver) should have come before getFunctionType(), and both getFunctionType() methods should be next to one another.